module-4/nmap/nmap.py

```diff
@@ -0,0 +1,80 @@
+import socket
+import argparse
+import concurrent.futures
+
+# Common services for the first 100 ports
+COMMON_PORTS = {
+    20: "FTP data",
+    21: "FTP control",
+    22: "SSH",
+    23: "Telnet",
+    25: "SMTP",
+    53: "DNS",
+    80: "HTTP",
+    110: "POP3",
+    143: "IMAP",
+    443: "HTTPS",
+    993: "IMAPS",
+    995: "POP3S",
+    # ... continue up to port 100 as needed
+}
+
+def resolve_host(host):
+    try:
+        ip = socket.gethostbyname(host)
+        print(f"[INFO] IP address for {host} is {ip}")
+        return ip
+    except socket.gaierror:
+        print(f"[ERROR] Could not resolve host {host}.")
+        return None
+
+def scan_port(ip, port):
+    with socket.socket(socket.AF_INET, socket.SOCK_STREAM) as s:
```

Comment:
please refrain from naming variables with just a single character like s.
This introduces many problems in software engineering in general, but foremost it's just lazy and a bad code-style pattern.

in the best case name it descriptive, so others can obtain information about the variable, function or similar, and everybody is able to know the purpose of this variable.